src/verbosity.jl

```diff
@@ -34,56 +42,57 @@ end
 
 "$vlogmacrodocs"
 macro debugv(verbosity::Int, msg, exs...)
-    return restore_callsite_source_position!(
+    return restore_callsite_source_position!(:debugv,
         esc(:($Base.@debug $msg _group=$(Verbosity(verbosity)) $(exs...))),
         __source__,
     )
 end
 
 "$vlogmacrodocs"
 macro infov(verbosity::Int, msg, exs...)
-    return restore_callsite_source_position!(
+    return restore_callsite_source_position!(:infov,
         esc(:($Base.@info $msg _group=$(Verbosity(verbosity)) $(exs...))),
         __source__,
     )
 end
 
 "$vlogmacrodocs"
 macro warnv(verbosity::Int, msg, exs...)
-    return restore_callsite_source_position!(
+    return restore_callsite_source_position!(:warnv,
         esc(:($Base.@warn $msg _group=$(Verbosity(verbosity)) $(exs...))),
         __source__,
     )
 end
 
 "$vlogmacrodocs"
 macro errorv(verbosity::Int, msg, exs...)
-    return restore_callsite_source_position!(
+    return restore_callsite_source_position!(:errorv,
         esc(:($Base.@error $msg _group=$(Verbosity(verbosity)) $(exs...))),
         __source__,
     )
 end
 
 "$vlogmacrodocs"
 macro logmsgv(verbosity::Int, level, msg, exs...)
-    return restore_callsite_source_position!(
+    return restore_callsite_source_position!(:logmsgv,
         esc(:($Base.@logmsg $level $msg _group=$(Verbosity(verbosity)) $(exs...))),
         __source__,
     )
 end
 
 """
-    LoggingExtras.withlevel(f, level; verbosity::Integer=0)
+    LoggingExtras.withlevel(f, level; verbosity::Integer=0, group::Union{Symbol, Nothing}=nothing)
 
 Convenience function like `Logging.with_logger` to temporarily wrap
 the current logger with a level filter while `f` is executed.
 That is, the current logger will still be used for actual logging, but
 log messages will first be checked that they meet the `level`
 log level before being passed on to be logged.
 
-For convenience, a `verbosity` keyword argument can be passed which also
-filters the "verbose logging" messages; see [`@debugv`](@ref), [`@infov`](@ref),
-[`@warnv`](@ref), [`@errorv`](@ref), and [`@logmsgv`](@ref).
+For convenience, a `group` keyword argument can be passed which also
+filters logging messages on the "group". By default, the group is the
+file name of the log macro call site, but can be overridden by passing
+the `_group` keyword argument to the logging macros.
```

Comment:
I am not conviced thsi is better than just using

```
with_logger(EarlyFilteredLogger(x->x.group==:Foo) do
...
end
```

Maybe we should instead be adding a 
```
filter_group_logger(group) = EarlyFilteredLogger(x->x.group==group)
```
if it really is too long

Reply:
It's really the convenience of having it bundled _with_ the _withlevel_ function. It's so onerous remembering the incantation to get the logger, set it to the new one, and then on top remember the LoggingExtras machinery to early filter. Bundling that all up into `withlevel` as the "one thing" I need to remember as a package dev is what makes this work really well.